### ISSUE TITLE
Kernel crng takes too long to initialize

### ISSUE THREAD

**Issue by bgilbert** [coreos/bugs#2419]
# Issue Report #

### Container Linux Version ###

```
$ cat /etc/os-release
NAME="Container Linux by CoreOS"
ID=coreos
VERSION=1758.0.0+2018-04-25-2334
VERSION_ID=1758.0.0
BUILD_ID=2018-04-25-2334
PRETTY_NAME="Container Linux by CoreOS 1758.0.0+2018-04-25-2334 (Rhyolite)"
ANSI_COLOR="38;5;75"
HOME_URL="https://coreos.com/"
BUG_REPORT_URL="https://issues.coreos.com"
COREOS_BOARD="amd64-usr"
```

### Environment ###

QEMU, AWS, others

### Expected Behavior ###

Boot proceeds with all due haste.

### Actual Behavior ###

`disk-uuid@.service` and `verity-setup.service` block for a long time (tens of seconds to minutes) during first boot.  Corresponding kernel logs:

```
[    3.039142] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
[  106.660290] random: crng init done
```

### Reproduction Steps ###

  1. `./coreos_production_qemu.sh`

### Other Information ###

This appears to have been introduced by coreos/linux@6e513bc20ca63f594632eca4e1968791240b8f18 in 4.14.x and coreos/linux@cd8d7a5778a4abf76ee8fe8f1bfcf78976029f8d in 4.16.x.  All Container Linux release _branches_ are affected, but no affected releases have been made on any channel.

**Comment by glevand** [comment]
Just some added info.  Both amd64 and arm64 are affected.  Git bisect of coreos-overlay gave me this:

````
000ac7f0bcdeeaabff80e8bbaf2bb985c75f144d is the first bad commit
commit 000ac7f0bcdeeaabff80e8bbaf2bb985c75f144d
Author: David Michael <david.michael@coreos.com>
Date:   Tue Apr 24 11:11:03 2018 -0400

    sys-kernel/coreos-sources: Bump 4.16.3 to 4.16.4
````

If you remove the infinite timeout on verity in bootengine:

 ````
--- a/dracut/10verity-generator/no-job-timeout.conf
+++b/dracut/10verity-generator/no-job-timeout.conf
@@ -1,2 +1,2 @@
 [Unit]
-JobRunningTimeoutSec=infinity
+JobRunningTimeoutSec=60
````

You'll see logs like these:

[ TIME ] Timed out waiting for device dev-mapper-usr.device.
[DEPEND] Dependency failed for Verity Setup for /dev/mapper/usr.
[DEPEND] Dependency failed for File System Check on /dev/mapper/usr.
[DEPEND] Dependency failed for /sysroot/usr.
[   66.929848] systemd[1]: dev-mapper-usr.device: Job dev-mapper-usr.device/start timed out.
[DEPEND] Dependency failed for Root filesystem setup.
[   66.937881] systemd[1]: Timed out waiting for device dev-mapper-usr.device.
[DEPEND] Dependency failed for Ignition (files).
[DEPEND] Dependency failed for Ignition (record completion).
[DEPEND] Dependency failed for Initrd Default Target.
[   66.941994] systemd[1]: Dependency failed for Verity Setup for /dev/mapper/usr.
[   66.944841] systemd[1]: verity-setup.service: Job verity-setup.service/start failed with result 'dependency'.
[   66.950260] systemd[1]: Dependency failed for File System Check on /dev/mapper/usr.
[   66.953222] systemd[1]: Dependency failed for /sysroot/usr.
[DEPEND] Dependency failed for Initrd File Systems.
[   66.957243] systemd[1]: Dependency failed for Root filesystem setup.
[   66.962282] systemd[1]: Dependency failed for Ignition (files).
[   66.963987] systemd[1]: Dependency failed for Ignition (record completion).
[   66.966863] systemd[1]: Dependency failed for Initrd Default Target.
[   66.972301] systemd[1]: initrd.target: Job initrd.target/start failed with result 'dependency'.
[   66.976433] systemd[1]: initrd.target: Triggering OnFailure= dependencies.
[   66.980446] systemd[1]: ignition-quench.service: Job ignition-quench.service/start failed with result 'dependency'.
[   66.984880] systemd[1]: ignition-files.service: Job ignition-files.service/start failed with result 'dependency'.
[   66.989290] systemd[1]: initrd-setup-root.service: Job initrd-setup-root.service/start failed with result 'dependency'.
[   66.994240] systemd[1]: Dependency failed for Initrd File Systems.
[   66.995968] systemd[1]: initrd-fs.target: Job initrd-fs.target/start failed with result 'dependency'.
[   67.001279] systemd[1]: initrd-fs.target: Triggering OnFailure= dependencies.
         Starting Setup Virtual Console...
[   67.005197] systemd[1]: sysroot-usr.mount: Job sysroot-usr.mount/start failed with result 'dependency'.
[   67.008990] systemd[1]: systemd-fsck@dev-mapper-usr.service: Job systemd-fsck@dev-mapper-usr.service/start failed with result 'dependency'.
[   67.013201] systemd[1]: dev-mapper-usr.device: Job dev-mapper-usr.device/start failed with result 'timeout'.
[  OK  ] Stopped dracut pre-mount hook.
[   67.019942] systemd[1]: Starting Setup Virtual Console...
[  OK  ] Stopped dracut initqueue hook.
[   67.029929] systemd[1]: Stopped dracut pre-mount hook.
[  OK  ] Stopped dracut pre-trigger hook.
[  OK  ] Stopped dracut pre-udev hook.
[  OK  ] Stopped dracut cmdline hook.
[   67.043649] systemd[1]: Stopped dracut initqueue hook.
[   67.051548] systemd[1]: Stopped dracut pre-trigger hook.
[   67.055668] systemd[1]: Stopped dracut pre-udev hook.
[   67.059389] systemd[1]: Stopped dracut cmdline hook.
[   67.125950] systemd-vconsole-setup[376]: KD_FONT_OP_GET failed while trying to get the font metadata: Function not implemented
[   67.131454] systemd-vconsole-setup[376]: Fonts will not be copied to remaining consoles
[   67.185654] systemd[1]: Started Setup Virtual Console.
[  OK  ] Started Setup Virtual Console.
[   67.228565] systemd[1]: Starting Emergency Shell...
         Starting Emergency Shell...

**Comment by glevand** [comment]
Just as a followup, if I revert these commits the boot works as expected.  

    e21e1049 sys-kernel/coreos-sources: Bump 4.16.5 to 4.16.6
    df2d002d sys-kernel/coreos-sources: Bump 4.16.4 to 4.16.5
    ff1b4db5 sys-kernel/coreos-kernel: use slotted dependency for intel-microcode
    000ac7f0 sys-kernel/coreos-sources: Bump 4.16.3 to 4.16.4

I pushed that out here:
https://github.com/glevand/coreos--coreos-overlay/commits/revert-to-kernel-4.16.3

**Comment by glevand** [comment]
Seems to be fixed with the upgrade to Linux-4.16.7.
